tests/installation/usb_install.pm

```diff
@@ -0,0 +1,111 @@
+    my $iso_usb = "/dev/$usb[0]";
+    my $ignition_usb = "/dev/$usb[1]";
+    record_info("Pick usb drive $iso_usb to store ISO and $ignition_usb to store ignition config.");
+
+    # Write ignition config to one usb
+    assert_script_run("echo y | mkfs.ext4 $ignition_usb", 60);
+    assert_script_run("e2label $ignition_usb ignition");
+    assert_script_run("mkdir  -p /mnt");
+    assert_script_run("mount $ignition_usb /mnt");
+    assert_script_run("mkdir -p /mnt/ignition");
+    my $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/ignition/config.ign")
+      . " -o /mnt/ignition/config.ign";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    assert_script_run("cat /mnt/ignition/config.ign");
+    save_screenshot;
+
+    # Write combustion script to the same usb with ignition
+    assert_script_run("mkdir -p /mnt/combustion");
+    $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/combustion/script")
+      . " -o /mnt/combustion/script";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    my $SERIALCONSOLE = get_required_var('SERIALCONSOLE');
+    assert_script_run("sed -i 's/SERIALCONSOLE/$SERIALCONSOLE/g' /mnt/combustion/script");
+    assert_script_run("chmod a+x /mnt/combustion/script");
+    assert_script_run("ls -l /mnt/combustion/script && cat /mnt/combustion/script");
+    save_screenshot;
+
+    # Flush to usb stick
+    assert_script_run("sync");
+    assert_script_run("umount -l /mnt");
+    record_info("Ignition and combustion files are successfully downloaded and written to usb $ignition_usb.");
+
+    # Download and dd the iso to the other usb
+    my $download_url = "http://" . get_var('OPENQA_URL', get_var('OPENQA_HOSTNAME')) . "/assets/iso/" . get_required_var('ISO');
+    die "ISO URL is not accessible: $download_url." unless head($download_url);
+    $cmd = "curl -L $download_url | dd of=$iso_usb bs=1M";
+    script_retry($cmd, retry => 2, delay => 10, timeout => 600, die => 1);
+    save_screenshot;
+
+    # Flush
+    assert_script_run("sync");
+    record_info("ISO is saved successfully to usb $iso_usb.");
+
+
+    # Set next boot to usb
+    if (check_var('IPXE_UEFI', '1')) {
+        # Some machines do not support setting floppy boot via ipmitool in uefi boot mode,
+        # so we use efibootmgr instead
+        record_info('efibootmgr output after dd iso to usb:', script_output('efibootmgr'));
+        save_screenshot;
+        # Some machines are problematic which show faulty boot entries as active boot.
+        # Add workaround USB_BOOT_DEV_KEYWORD to filter correct boot.
+        my $cmd = 'efibootmgr | grep usb -i | grep -v generic -i | grep "\*"';
+        $cmd .= get_var('USB_BOOT_DEV_KEYWORD') ? ' | grep ' . get_var('USB_BOOT_DEV_KEYWORD') : '';
+        my $usb_boot = script_output("$cmd");
```

Comment:
```$usb_boot``` looks like as below on my laptop:

```
localhost:/ # efibootmgr | grep usb -i | grep -v generic -i | grep "\*"
Boot001A* USB CD
Boot001B* USB FDD
Boot0020* USB HDD
```

Reply:
That's why USB_BOOT_DEV_KEYWORD is  introduced ;-)

Reply:
I do not think the code finding bootable usb device here is good enough because it can not even find the most common ubs boot device on a ordinary laptop. I think the code here should be improved, for example, finding bootable usb device with more reliable and common pattern.

Reply:
>  because it can not even find the most common ubs boot device on a ordinary laptop
I do not quite get what you mean here.

Besides, would you please give explicitly what way you recommend to use?

Reply:
Your code responsible for finding bootable usb device should be effective and reliable to find such device, at least, on most common test server instead of only relying on USB_BOOT_DEV_KEYWORD. Otherwise your code become ineffective.

Reply:
I do not find the common way you mentioned, and to be honest, I still do not know what is the "common" way in your suggestion. So again, if you have a specific method, welcome to share. Otherwise this will be the one.

Reply:
Since no more suggestion on this, mark it resolved now.